RMET-3602 ::: Update sqlcipher for 16KB page alignment

OS-pedrogustavobilro opened the pull request:
## Description

Update the sqlcipher library [to the latest version](https://discuss.zetetic.net/t/android-sqlcipher-support-for-android-sdk-35-16k-page-sizes/6460/11) - with compatibility for 16KB page size.

Instead of having [a custom aar](https://github.com/OutSystems/android-database-sqlcipher/tree/outsystems) that had a fix for API level 23 (no longer needed), now we get it the sqlcipher community version directly. Required some minor changes as the older library - [as described here](https://www.zetetic.net/sqlcipher/sqlcipher-for-android-migration/).

### Note

As of 12-Nov-2024 - A MABS 10 build cannot be installed in a device with 16KB page size enabled. However if we download the source and update the Android Gradle Plugin from 8.2.0 to 8.3.0 it can be installed. And runs fine.

The AGP version must be set by MABS, so MABS 10 would need to be updated for the builds to be installed there (and with an updated sqlcipher version if it hasn't been updated already).

## Context

- https://outsystemsrd.atlassian.net/browse/RMET-3602
- https://outsystemsrd.atlassian.net/wiki/spaces/RDME/pages/4426137659/iOS+18+Android+15+Assessment
- https://developer.android.com/guide/practices/page-sizes#build

## Type of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply -->
- [X] Fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [X] Android
- [ ] iOS
- [ ] JavaScript

## Tests
<!--- Describe how you tested your changes in detail -->
<!--- Include details of your test environment if relevant -->

### Run in 16KB page size emulator

1. Create an Android 15 emulator with a system image that has 16kb page size
2. Generate a MABS11 build of ciphered sample app with the plugin pointing to the branch https://github.com/OutSystems/cordova-outsystems-secure-sqlite-bundle/tree/fix/RMET-3602/support-16KB-page-size
3. Install and run that build
4. The app should not crash anymore.

### Update from an older version

1. Have an older version that is not pointed to this MR - e.g. there is a MABS 11 build from 12th Nov 2024 at 11:41
2. Install and run that app on any device (except one with 16KB page size enabled - that will not work)
3. Add some Employees to the DB
4. Update the app to a MABS11 build with barcode plugin pointing to the branch https://github.com/OutSystems/cordova-outsystems-secure-sqlite-bundle/tree/fix/RMET-3602/support-16KB-page-size
5. Confirm that the DB is still accessible and correct after updating the app

### Check Page alignment

Steps:

1. Generate a MABS11 build of ciphered sample app with the plugin pointing to the branch https://github.com/OutSystems/cordova-outsystems-secure-sqlite-bundle/tree/fix/RMET-3602/support-16KB-page-size
6. Run `./check_elf_alignment.sh <apk_file> | grep arm64-v8a` - ".so" files associated with sqlcipher should be aligned
```
=== ELF alignment ===
/var/folders/ln/4t6dd54d3c1b3wm92vv9pggc0000gn/T/Ciphered_Local_Storage_Sample_App_update_sqlcipher_out_XXXXX.4tWJCRlKn2/lib/armeabi-v7a/libsqlcipher.so: \e[32mALIGNED\e[0m (2**14)
/var/folders/ln/4t6dd54d3c1b3wm92vv9pggc0000gn/T/Ciphered_Local_Storage_Sample_App_update_sqlcipher_out_XXXXX.4tWJCRlKn2/lib/x86/libsqlcipher.so: \e[32mALIGNED\e[0m (2**14)
/var/folders/ln/4t6dd54d3c1b3wm92vv9pggc0000gn/T/Ciphered_Local_Storage_Sample_App_update_sqlcipher_out_XXXXX.4tWJCRlKn2/lib/arm64-v8a/libsqlcipher.so: \e[32mALIGNED\e[0m (2**14)
/var/folders/ln/4t6dd54d3c1b3wm92vv9pggc0000gn/T/Ciphered_Local_Storage_Sample_App_update_sqlcipher_out_XXXXX.4tWJCRlKn2/lib/x86_64/libsqlcipher.so: \e[32mALIGNED\e[0m (2**14)
ELF Verification Successful
=====================
```

## Checklist
<!--- Go over all the following items and put an `x` in all the boxes that apply -->
- [ ] Pull request title follows the format `RMET-XXXX <title>`
- [ ] Code follows code style of this project
- [ ] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

OS-pedrogustavobilro committed:
fix: Update sqlcypher to new location and latest version

References:

- https://outsystemsrd.atlassian.net/browse/RMET-3602
- https://outsystemsrd.atlassian.net/wiki/spaces/RDME/pages/4426137659/iOS+18+Android+15+Assessment

OS-pedrogustavobilro committed:
fix: Add missing sqlite DB version

References:

- https://outsystemsrd.atlassian.net/browse/RMET-3602
- https://outsystemsrd.atlassian.net/wiki/spaces/RDME/pages/4426137659/iOS+18+Android+15+Assessment

OS-pedrogustavobilro committed:
Revert "fix: Add missing sqlite DB version"

This reverts commit 849e78dd0bd9c9cb247d1ebee1b99e5e5e5eb096.

OS-pedrogustavobilro committed:
fix: Changes related to migrating sqlcipher library

References:

- https://www.zetetic.net/sqlcipher/sqlcipher-for-android-migration/
- https://outsystemsrd.atlassian.net/browse/RMET-3602
- https://outsystemsrd.atlassian.net/wiki/spaces/RDME/pages/4426137659/iOS+18+Android+15+Assessment

Chuckytuh commented on plugin.xml:
```diff
@@ -37,7 +37,7 @@
         <source-file src="src/android/io/sqlc/SQLitePlugin.java" target-dir="src/io/sqlc"/>
         <source-file src="src/android/io/sqlc/SQLiteAndroidDatabase.java" target-dir="src/io/sqlc"/>
 
-        <lib-file src="src/android/libs/android-database-sqlcipher-4.5.4-custom.aar" />
```
If the aar is not being used anymore, shouldn't it be deleted from the repo?

OS-pedrogustavobilro replied:
Correct yes, thanks for pointing it out!

Done in https://github.com/OutSystems/Cordova-sqlcipher-adapter/pull/14/commits/0739cbef5ff4d9d30e87c0a69e0f7ce6c83476b6

OS-pedrogustavobilro committed:
chore: Remove unused custom aar

References:

- https://www.zetetic.net/sqlcipher/sqlcipher-for-android-migration/
- https://outsystemsrd.atlassian.net/browse/RMET-3602
- https://outsystemsrd.atlassian.net/wiki/spaces/RDME/pages/4426137659/iOS+18+Android+15+Assessment

OS-pedrogustavobilro committed:
chore: Update CHANGES.md

References:

- https://outsystemsrd.atlassian.net/browse/RMET-3602
- https://outsystemsrd.atlassian.net/wiki/spaces/RDME/pages/4426137659/iOS+18+Android+15+Assessment

OS-pedrogustavobilro committed:
fix: Add missing mavencentral repo for sqlcipher dependency

References:

- https://outsystemsrd.atlassian.net/browse/RMET-3602
- https://outsystemsrd.atlassian.net/wiki/spaces/RDME/pages/4426137659/iOS+18+Android+15+Assessment

alexgerardojacinto committed:
chore(release): raise to version 0.1.7-OS9

References: https://outsystemsrd.atlassian.net/browse/RMET-3844
